magefiles/sprayproxy.go

```diff
@@ -0,0 +1,170 @@
+		HTTPClient: &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					// #nosec G402
+					InsecureSkipVerify: true,
+				},
+			},
+		},
+	}
+}
+
+func (s *SprayProxy) RegisterServer(hostUrl string) (string, error) {
+	result, err := s.sendRequest(hostUrl, http.MethodPost)
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+func (s *SprayProxy) UnegisterServer(hostUrl string) (string, error) {
+	result, err := s.sendRequest(hostUrl, http.MethodDelete)
+	if err != nil {
+		return "", err
+	}
+	return result, nil
+}
+
+func (s *SprayProxy) GetServers() (string, error) {
+	result, err := s.sendRequest("", http.MethodGet)
+	if err != nil {
+		return "", err
+	}
+	return result, nil
+}
+
+func (s *SprayProxy) sendRequest(hostUrl string, httpMethod string) (string, error) {
+	requestURL := s.BaseURL + "/backends"
+
+	data := make(map[string]string)
+	data["url"] = hostUrl
+	bytesData, _ := json.Marshal(data)
+
+	req, err := http.NewRequest(httpMethod, requestURL, bytes.NewReader(bytesData))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", s.Token))
+	
+	res, err := s.HTTPClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if res.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to access SprayProxy server with status code: %d and\nbody: %s", res.StatusCode, string(body))
+	}
+
+	defer res.Body.Close()
+
+	return string(body), err
+}
+
+func GetPod() (*corev1.Pod, error) {
+	k8sClient, err := kubeCl.NewAdminKubernetesClient()
+	if err != nil {
+		return nil, err
+	}
+	podList, err := k8sClient.KubeInterface().CoreV1().Pods(SPRAYPROXY_NAMESPACE).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			return &pod, nil
+		}
+	}
+	return nil, fmt.Errorf("no running pod found in namespace %s", SPRAYPROXY_NAMESPACE)
+}
+
+func GetSprayProxyToken() (token string, err error){
+	pod, err := GetPod()
+	if err != nil {
+		return "", err
+	}
+	// #nosec G204
+	tokenBytes, err := exec.Command("oc", "exec", "-n", SPRAYPROXY_NAMESPACE, pod.Name, "-c", "kube-rbac-proxy", "-i", "-t", "--", "cat", "/var/run/secrets/kubernetes.io/serviceaccount/token").Output()
+	if err != nil {
+		klog.Errorf("error obtaining oc token %s", err.Error())
+	}
+	return strings.TrimSuffix(string(tokenBytes), "\n"), nil
+}
+
+func GetSprayProxyRoute() (*routev1.Route, error) {
+	k8sClient, err := kubeCl.NewAdminKubernetesClient()
+	if err != nil {
+		return nil, err
+	}
+
+	namespaceName := types.NamespacedName{
+		Name:      ROUTE_NAME,
+		Namespace: SPRAYPROXY_NAMESPACE,
+	}
+
+	route := &routev1.Route{}
+	err = k8sClient.KubeRest().Get(context.TODO(), namespaceName, route)
+	if err != nil {
+		return &routev1.Route{}, err
+	}
+	return route, nil
+}
```

Comment:
I tried to verify this locally with my github app id, but it didn't work - it created the PR, but no pipelinerun was triggered.

I don't see these functions being used anywhere - could it be caused by that? should sprayproxy route be set as a webhook URL in a github app? I think we need this step to be automated

Reply:
It doesn't need the above any more. Because when you set up it locally, it doesn't need to deploy a sprayproxy as the global sprayprox.  If you agree, I will remove them .

Reply:
yep agree to remove them

Reply:
fixed